fvm/programs/block_programs.go

```diff
@@ -118,12 +154,47 @@ func (transaction *DerivedTransactionData) AddInvalidator(
 	invalidator TransactionInvalidator,
 ) {
 	transaction.programs.AddInvalidator(invalidator.ProgramInvalidator())
+	transaction.meterParamOverrides.AddInvalidator(
+		invalidator.MeterParamOverridesInvalidator())
+}
+
+func (transaction *DerivedTransactionData) GetMeterParamOverrides(
+	txnState *state.TransactionState,
+	getMeterParamOverrides ValueComputer[int, MeterParamOverrides],
+) (
+	MeterParamOverrides,
+	error,
+) {
+	return transaction.meterParamOverrides.GetOrCompute(
+		txnState,
+		meterParamOverridesKey,
+		getMeterParamOverrides)
 }
 
 func (transaction *DerivedTransactionData) Validate() RetryableError {
-	return transaction.programs.Validate()
+	err := transaction.programs.Validate()
+	if err != nil {
+		return err
+	}
+
+	err = transaction.meterParamOverrides.Validate()
+	if err != nil {
+		return err
+	}
+
+	return nil
 }
 
 func (transaction *DerivedTransactionData) Commit() RetryableError {
-	return transaction.programs.Commit()
+	err := transaction.programs.Commit()
+	if err != nil {
+		return err
+	}
+
+	err = transaction.meterParamOverrides.Commit()
+	if err != nil {
+		return err
```

Comment:
better to add error message to distinguish the error

Reply:
good point.  will do